profiler/profiler.go

```diff
@@ -173,14 +176,17 @@ func (p *profiler) collect(ticker <-chan time.Time) {
 				// configured CPU profile duration: (start-end).
 				end: now.Add(p.cfg.cpuDuration),
 			}
-			for t := range p.cfg.types {
-				prof, err := p.runProfile(t)
+
+			for _, t := range p.enabledProfileTypes() {
```

Comment:
Can the result of this function be changed at runtime? If not, should we just get the list before we start the `for` loop?

Reply:
> Can the result of this function be changed at runtime?

No, I don't think so right now.

> If not, should we just get the list before we start the for loop?

What's your concern? Performance? I'd rather optimize for simplicity here and avoid additional indirection.

Reply:
I don't think that makes it any less simple.